extension/content/components/recipes/form/arguments/BranchedAddon.js

```diff
@@ -0,0 +1,218 @@
+// @ts-nocheck
+
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  Col,
+  ControlLabel,
+  Divider,
+  FormGroup,
+  HelpBlock,
+  Icon,
+  IconButton,
+  Input,
+  InputNumber,
+  Panel,
+  Row,
+  SelectPicker,
+} from "rsuite";
+
+import {
+  ACTION_UPDATE_DATA,
+  useRecipeDetailsData,
+  useRecipeDetailsDispatch,
+} from "devtools/contexts/recipeDetails";
+import {
+  useEnvironmentState,
+  useSelectedNormandyEnvironmentAPI,
+} from "devtools/contexts/environment";
+import InputField from "devtools/components/recipes/form/arguments/fields/InputField";
+import ToggleField from "devtools/components/recipes/form/arguments/fields/ToggleField";
+
+export default function BranchedAddon() {
+  const { selectedKey: environmentKey } = useEnvironmentState();
+  const [extensions, setExtensions] = React.useState([]);
+  const normandyApi = useSelectedNormandyEnvironmentAPI();
+
+  React.useEffect(() => {
+    normandyApi.fetchAllExtensions().then((allExtensions) => {
+      setExtensions(allExtensions);
+    });
+  }, [environmentKey]);
+
+  const options = extensions.map((extension) => ({
+    label: extension.name,
+    value: extension.id,
+  }));
```

Comment:
This should actually probably move to `Branches` and then pass to each `Branch` as `extensionOptions` so it's clear what they are options for.